src/main/kotlin/uk/gov/justice/digital/hmpps/jobsboard/api/jobs/application/ExpressionOfInterestEditor.kt

```diff
@@ -0,0 +1,56 @@
+package uk.gov.justice.digital.hmpps.jobsboard.api.jobs.application
+
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.jobsboard.api.commons.domain.EntityId
+import uk.gov.justice.digital.hmpps.jobsboard.api.jobs.domain.ExpressionOfInterest
+import uk.gov.justice.digital.hmpps.jobsboard.api.jobs.domain.ExpressionOfInterestId
+import uk.gov.justice.digital.hmpps.jobsboard.api.jobs.domain.ExpressionOfInterestRepository
+import uk.gov.justice.digital.hmpps.jobsboard.api.jobs.domain.JobRepository
+
+@Service
+class ExpressionOfInterestEditor(
+  private val jobRepository: JobRepository,
+  private val expressionOfInterestRepository: ExpressionOfInterestRepository,
+) {
+
+  @Transactional
+  fun createWhenNotExist(request: CreateOrDeleteExpressionOfInterestRequest): Boolean {
```

Comment:
I can see that you decided to use a different name than the one at Jobs. To be honest, this was one of the tough ones when I created it in Jobs. I used the name 'createOrUpdate' for this action because understood that it will try to update the record if it is already created.

I understand that the 'update' word can be misunderstood because it implies a change, and we don't have any parameter changes. But the action that we do here when we set again an expression of interest is called an update.

I prefer to be consistent with the rest of the code and call this function 'createOrUpdate' and, if we want, change all occurrences to any new name that we consider better.

Reply:
let me split create, delete for cleanness.
I named it this way for sharing with delete

Reply:
updated

Reply:
ah, I forgot to highlight this method is a bit different from normal createOrUpdate(): there is a return value.

This method return "created" flag to determine pre-existence, such that REST controller can reply 201 or 200

Reply:
I remember thinking about that with the Job creation functionality and deciding to create a separate function called `existsById(jobId)`, which returns true if it exists and false if not. I did that because it was not clear, reading the name of the function creation function, when it would return `true` or `false`.

Now, revisiting this code, I can also find that maybe the best place to put that function is in the `JobRetriever` instead of in the `JobCreator` class.

The functionality is the same: you do fire and forget, and the controller's response is managed depending on whether that record exists or not.

Reply:
Updated again, please re-review